frontend/src/lib/WorkflowParser.ts

```diff
@@ -75,6 +75,13 @@ export default class WorkflowParser {
         statusMap.set(status[taskRunId]['pipelineTaskName'], status[taskRunId]);
     }
 
+    // Add custom tasks to the status map
+    for (const taskRunId of Object.getOwnPropertyNames(workflow['status']['runs'])) {
```

Comment:
Object.getOwnPropertyNames() may return nil instead of empty array. E.g. you will get an error when running the cache example. https://github.com/kubeflow/kfp-tekton/blob/master/sdk/python/tests/compiler/testdata/cache.py

Reply:
need to fall back to an empty array if Object.getOwnPropertyNames() == nil